### PR TITLE
Backend - Added modern time formatters

### PR DESCRIPTION
**Pull request by Ark-kun**
Placeholder format: `{{$.scheduledTime.strftime('%Y-%m-%d')}}`

### CODE REVIEW

**Comment by Bobgy** [comment]
@Ark-kun thank you! This is awesome 😎

Do you need any help finishing this?

**Comment by Ark-kun** [comment]
Sorry for the delay. Fixed the regex bug.
The backend test now passes.

**Comment by Bobgy** [comment]
Thank you so much!
I will review next week

File: backend/src/common/util/parameter_formatter.go [hunk 1]
```diff
@@ -113,6 +119,22 @@ func (p *ParameterFormatter) createSubstitutes(match string) string {
...
+		formatter, err := strftime.New(format, strftime.WithUnixSeconds('s'))
+		if err != nil {
+			return match
+		}
+		return formatter.FormatString(time.Unix(p.scheduledEpoch, 0).UTC())
+	} else if p.nowEpoch != disabledField && strings.HasPrefix(match, currentTimePrefix2) {
+		format := strings.Replace(match, currentTimePrefix2, "", 1)
+		format = strings.Replace(format, suffix2, "", 1)
+		formatter, err := strftime.New(format, strftime.WithUnixSeconds('s'))
+		if err != nil {
```
Comment (Bobgy) [inline backend/src/common/util/parameter_formatter.go:134]: Log a warning?

File: backend/src/common/util/parameter_formatter.go [hunk 2]
```diff
@@ -113,6 +119,22 @@ func (p *ParameterFormatter) createSubstitutes(match string) string {
...
 		match = strings.Replace(match, suffix, "", 1)
 		return time.Unix(p.nowEpoch, 0).UTC().Format(match)
+	} else if p.scheduledEpoch != disabledField && strings.HasPrefix(match, scheduledTimePrefix2) {
+		format := strings.Replace(match, scheduledTimePrefix2, "", 1)
+		format = strings.Replace(format, suffix2, "", 1)
+		formatter, err := strftime.New(format, strftime.WithUnixSeconds('s'))
+		if err != nil {
+			return match
+		}
+		return formatter.FormatString(time.Unix(p.scheduledEpoch, 0).UTC())
```
[the remark this answers is not in the document]
Comment (Ark-kun) [reply backend/src/common/util/parameter_formatter.go:129]: This PR follows the rules for the existing formatters.
We can make another PR to use the timezone config everywhere in this module.

File: backend/src/common/util/parameter_formatter.go [hunk 3]
```diff
@@ -113,6 +119,22 @@ func (p *ParameterFormatter) createSubstitutes(match string) string {
 		match = strings.Replace(match, currentTimePrefix, "", 1)
 		match = strings.Replace(match, suffix, "", 1)
 		return time.Unix(p.nowEpoch, 0).UTC().Format(match)
+	} else if p.scheduledEpoch != disabledField && strings.HasPrefix(match, scheduledTimePrefix2) {
+		format := strings.Replace(match, scheduledTimePrefix2, "", 1)
+		format = strings.Replace(format, suffix2, "", 1)
+		formatter, err := strftime.New(format, strftime.WithUnixSeconds('s'))
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (Ark-kun) [reply backend/src/common/util/parameter_formatter.go:127]: Done.

File: backend/src/common/util/parameter_formatter.go (hunk 1 above)
Comment (Ark-kun) [reply backend/src/common/util/parameter_formatter.go:136]: Done.

**Comment by Bobgy** [comment]
/lgtm
/approve
Good point, we can change time zone in a separate PR

**Pull request merged**